sdk/core/azure-core/src/main/java/com/azure/core/util/ConfigurationPropertyBuilder.java

```diff
@@ -0,0 +1,137 @@
+     *
+     * @param shared If set to {@code true}, {@link Configuration#get(ConfigurationProperty)} will attempt to retrieve property from
+     *               local configuration and fall back to shared section, when local property is missing.
+     *               Otherwise, only local configuration will be checked.
+     * @return the updated ConfigurationPropertyBuilder object.
+     */
+    public ConfigurationPropertyBuilder<T> shared(boolean shared) {
+        this.shared = shared;
+        return this;
+    }
+
+    /**
+     * Sets flag indicating if property value can be logged.
+     * Default is {@code false}, indicating that property value will be redacted in logs.
+     *
+     * @param canLogValue If set to {@code true}, {@link Configuration#get(ConfigurationProperty)} will log property value,
+     *                    Otherwise, value is redacted.
+     * @return the updated ConfigurationPropertyBuilder object.
+     */
+    public ConfigurationPropertyBuilder<T> canLogValue(boolean canLogValue) {
+        if (canLogValue) {
+            this.valueSanitizer = PERMIT_VALUE_SANITIZER;
+        }
+
+        return this;
+    }
+
+    /**
+     * Sets flag indicating if property is required.
+     * Default is {@code false}, indicating that property is optional.
+     *
+     * @param required If set to {@code true}, {@link Configuration#get(ConfigurationProperty)} will throw when property is not found.
+     * @return the updated ConfigurationPropertyBuilder object.
+     */
+    public ConfigurationPropertyBuilder<T> required(boolean required) {
+        this.required = required;
+        return this;
+    }
+
+    /**
+     * Sets one or more alias for property. {@link Configuration#get(ConfigurationProperty)} attempts to retrieve property by name first
+     * and only then tries to retrieve properties by alias in the order aliases are provided.
+     *
+     * @param aliases one or more alias for the property name.
+     * @return the updated ConfigurationPropertyBuilder object.
+     */
+    public ConfigurationPropertyBuilder<T> aliases(String... aliases) {
+        this.aliases = aliases;
+        return this;
+    }
+
+    /**
+     * Sets one or more environment variable (or system property) property name.
+     * {@link Configuration#get(ConfigurationProperty)} falls back when property value is not found by {@code name}, alias,
+     * in local and shared configuration (if enabled) and only then checks environment variables (and system properties).
+     * When environment properties are not set, {@link Configuration#get(ConfigurationProperty)} does not attempt to
+     * read environment configuration.
+     *
+     * @param environmentAliases one or more environment variable (or system property).
+     * @return the updated ConfigurationPropertyBuilder object.
+     */
+    public ConfigurationPropertyBuilder<T> environmentAliases(String... environmentAliases) {
```

Comment:
Do we need both aliases and environmentAliases?

Reply:
I separate them because I don't want users to mix env vars/sys properties and other configuration options, e.g.  I don't want `AZURE_CLIENT_ID` to work if it's set in the config file (the opposite is also true: `http.proxy.hostname` should not work if set as env var).

Also during the arch board review, I believe, one of the points was that env vars are consistent across languages - we probably want to grow the list of supported env vars slowly.